File: InsuranceCompany.Core/Models/InsuranceRequest.cs
using InsuranceCompany.Core.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace InsuranceCompany.Core;

public partial class InsuranceRequest
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id { get; set; }
    public DateTime? DateOfStart { get; set; }
    public DateTime? DateOfEnd { get; set; }
    [ForeignKey(nameof(Agent))]
    public Guid? AgentId { get; set; }
    [ForeignKey(nameof(InsuranceRate))]
    public Guid? InsuranceRateId { get; set; }
    [ForeignKey(nameof(InsuranceStatus))]
    public Guid? InsuranceStatusId { get; set; }
    public virtual Agent? Agent { get; set; }
    public decimal Cost { get; set; }
    public decimal? BasePayment { get; set; }
    public decimal? UnitPayment { get; set; }
    public decimal? Benefits { get; set; }
    public decimal? Coefficient { get; set; }
    public bool IsReadyDocuments { get; set; }
    public virtual ICollection<AnswerValue> AnswerValues { get; } = new List<AnswerValue>();
    public virtual ICollection<InsuredPerson> InsuredPersons { get; } = new List<InsuredPerson>();
    public virtual ICollection<Document> Documents { get; } = new List<Document>();
    public virtual InsuranceRate? InsuranceRate { get; set; }
    public virtual InsuranceStatus? InsuranceStatus { get; set; }
}

File: InsuranceCompany.Client/src/app/models/InsuranceRequest.ts
import { InsuranceStatus } from 'src/app/models/InsuranceStatus';
import { Client } from 'src/app/models/Client';
import { InsuranceRate } from 'src/app/models/InsuranceRate';
import { InsuredPerson } from 'src/app/models/InsuredPerson';
import { AnswerValues } from 'src/app/models/AnswerValue';
import { Document } from './Document';

export class InsuranceRequest {
    id : string;
    dateOfStart : Date;
    dateOfEnd : Date;
    agentId : string;
    insuranceRateId : string;
    insuranceStatusId : string;
    mainClient : Client;
    insuranceStatus: InsuranceStatus;
    cost: number;
    сoefficient: number;
    benefits: number;
    basePayment: number;
    unitPayment: number;
    insuranceRate: InsuranceRate;
    insuredPersons: InsuredPerson[];
    answerValues: AnswerValues[];
    documents: Document[];
    isReadyDocuments: boolean;


    constructor(id: string, dateOfStart: Date, dateOfEnd: Date, agentId: string, 
        insuranceRateId: string, insuranceStatusId: string, insuranceStatus: InsuranceStatus, mainClient : Client,
        cost: number) 
    {
        this.id = id;
        this.dateOfStart = dateOfStart;
        this.dateOfEnd = dateOfEnd;
        this.agentId = agentId;
        this.insuranceRateId = insuranceRateId;
        this.insuranceStatusId = insuranceStatusId;
        this.insuranceStatus = insuranceStatus;
        this.mainClient = mainClient;
        this.cost = cost;
    }
}